src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -36,6 +36,28 @@ public function checkIfFieldNameBelongsToAnyRelation($fieldName)
         return false;
     }
 
+    /**
+     * Invokes $method_name in $model and returns true in case the return of the method call
+     * is an instance of Illuminate\Database\Eloquent\Relations\Relation.
+     *
+     * @param string $method_name
+     * @return bool
+     */
+    protected function checkIfMethodReturnRelation($method_name)
+    {
+        try {
+            $return = $this->model->{$method_name}();
+
+            if ($return instanceof Relation) {
+                return true;
+            }
+
+            return false;
+        } catch (\Exception $e) {
+            return false;
+        }
+    }
```

Comment:
I've written this before the comment above, where I propose we DO call the method.
Which stands to show how wild an opinion can vary, within 1-2 hours.

Of course, we should never EVER call all methods, to see if the have return types. EVER. But maybe we _could_ just call the method that was passed as the ```name``` of the field...

Food for thought.

Reply:
In this case though - if we do call the method that was passed as the name of the field. I don't think there's any point in checking its return type. It would do us no good. We call it and that's it.

Reply:
> I've written this before the comment above, where I propose we DO call the method.
> Which stands to show how wild an opinion can vary, within 1-2 hours.
> 
> Of course, we should never EVER call all methods, to see if the have return types. EVER. But maybe we _could_ just call the method that was passed as the `name` of the field...
> 
> Food for thought.

Atm we are not calling every method. We are just checking methods return types without calling them. 

```php
try {
            $reflect = new \ReflectionClass($model);

            foreach ($reflect->getMethods(\ReflectionMethod::IS_PUBLIC) as $method) {
                if ($method->hasReturnType()) {
                    $returnType = $method->getReturnType();
                    if (in_array(class_basename($returnType->getName()), $eloquentRelationships)) {
//only here, if it's return type is a relationship 
//we call it in getFieldAttributesFromRelationship();
                        $relations[] = $this->getFieldAttributesFromRelationship($model, $method->getName());
                    }
                }
            }
            return $relations;

```